Automattic/jetpack
# Nav Unification: Store the correct preferred view when it is selected.

Fixes https://github.com/Automattic/wp-calypso/issues/49486

#### Changes proposed in this Pull Request:
 
* This PR prevents preferred views for the same page being stored twice, without the query params and with the query params. This caused navigation from post to go to classic view when the user is expecting the new default/calypso view.
This is accomplished by removing the query args before saving the view. 

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Repro the issue: 
          1. Go to post and change view to classic view
          2. View post in the editor then use the W > View all posts to navigate back
          3. Change the view to default view then open the post editor
          4. Again navigate back, notice it's now showing the classic view when it should be showing the default view.

* Sandbox the generated diff D84120-code
* Perform the steps above and verify the correct view is being displayed.

For regression see here to test other screens 715-gh-Automattic/wpcomsh
